diff --git a/node_modules/vuepress-theme-gungnir/lib/client/components/NavLink.vue b/node_modules/vuepress-theme-gungnir/lib/client/components/NavLink.vue
index fe66da1..2031a7e 100644
--- a/node_modules/vuepress-theme-gungnir/lib/client/components/NavLink.vue
+++ b/node_modules/vuepress-theme-gungnir/lib/client/components/NavLink.vue
@@ -110,7 +110,7 @@ const isActiveInSubpath = computed(() => {
   if (!shouldBeActiveInSubpath.value) {
     return false;
   }
-  return route.path.startsWith(item.value.link);
+  return route?.path.startsWith(item.value.link);
 });
 
 // if this link is active
@@ -119,7 +119,7 @@ const isActive = computed(() => {
     return false;
   }
   if (item.value.activeMatch) {
-    return new RegExp(item.value.activeMatch).test(route.path);
+    return new RegExp(item.value.activeMatch).test(route?.path);
   }
   return isActiveInSubpath.value;
 });
diff --git a/node_modules/vuepress-theme-gungnir/lib/client/components/Navbar.vue b/node_modules/vuepress-theme-gungnir/lib/client/components/Navbar.vue
index 387ea01..2e8a6b8 100644
--- a/node_modules/vuepress-theme-gungnir/lib/client/components/Navbar.vue
+++ b/node_modules/vuepress-theme-gungnir/lib/client/components/Navbar.vue
@@ -59,7 +59,7 @@ const router = useRouter();
 const navbar = ref<HTMLElement | null>(null);
 const siteBrand = ref<HTMLElement | null>(null);
 const siteBrandLink = computed(() => themeLocale.value.home || "/");
-const siteBrandTitle = "$ cd /home/";
+const siteBrandTitle = computed(() => themeLocale.value.homeText ?? "$ cd /home/");
 const linksWrapperMaxWidth = ref(0);
 const linksWrapperStyle = computed(() => {
   if (!linksWrapperMaxWidth.value) {
diff --git a/node_modules/vuepress-theme-gungnir/lib/client/layouts/404.vue b/node_modules/vuepress-theme-gungnir/lib/client/layouts/404.vue
index 2e09efa..2ab2b77 100644
--- a/node_modules/vuepress-theme-gungnir/lib/client/layouts/404.vue
+++ b/node_modules/vuepress-theme-gungnir/lib/client/layouts/404.vue
@@ -11,6 +11,7 @@
 </template>
 
 <script setup lang="ts">
+import { computed } from "vue";
 import { useRouteLocale } from "@vuepress/client";
 import Common from "../components/Common.vue";
 import { useThemeLocaleData } from "../composables";
@@ -21,7 +22,7 @@ const themeLocale = useThemeLocaleData();
 const messages = themeLocale.value.notFound ?? ["Not Found"];
 const getMsg = (): string =>
   messages[Math.floor(Math.random() * messages.length)];
-const homeText = themeLocale.value.backToHome ?? "$ cd /home/";
+const homeText = computed(() => themeLocale.value.backToHome ?? "$ cd /home/");
 
 const emojiArray = [
   "\\(o_o)/",
